client/src/js/services/grid/GridColumns.js

```diff
@@ -85,7 +87,10 @@ function GridColumnService(uiGridConstants, AppCache, Modal, util) {
   function Columns(gridOptions, cacheKey) {
 
     // set up local storage for selected grid columns
-    this.cache = AppCache(cacheKey + serviceKey);
+    if(cacheKey){
```

Comment:
Just to make the service more flexible, because a grid should decide if his columns visibility state must be persistent or not.
For the trial balance grid for example, I don't need it but is important for posting journal and general ledger though.
That why grouping functionality is kind to plugin to the grid.